action.yaml

```diff
@@ -6,52 +6,75 @@ description:
   found to be resolved, allows the previously added comment, if one exists at
   that time, to be removed from the PR.
 inputs:
-  instruction-text:
+  pull-request-number:
+    required: true
+    description: Uniquely identifies the Pull Request of interest.
+  comment:
     required: true
     description:
       This is the full text of the message to be placed within a comment of the
       given PR to instruct the author of the PR how to resolve a given problem.
-  instruction-text-detector:
+  text-detector:
     required: true
     description:
-      This is some unique verbatim subset of the instruction-text that is to be
-      used to determine if a comment has already been created against the PR
-      that instructs the author how to resolve the given problem.
-  delete-comment-on-resolution:
+      This is some unique verbatim subset of the comment that is to be used to
+      determine if a comment has already been created against the PR that
+      instructs the author how to resolve the given problem.
+    default: created by action-conditional-pr-comment
+  add-or-remove-comment:
     required: true
     description:
-      Whether or not to delete the comment that has been added to the PR upon
-      problem resolution.
-    default: "true"
+      true indicates the comment is to be added/updated to/within the PR, false
+      indicates the comment is to be removed from the PR.
```

Comment:
this name is a tricky one.  

I guess that we want to be able to pass in a `boolean` check?

Naming wise `add or remove` doesn't map well to a boolean.  I don't know if I have a better suggestion.

Reply:
True, and having two separate booleans would just add more confusion. One idea is to accept a `workflow` input that expects either `ADD_UPDATE` or `REMOVE`.

Reply:
I have refactored to the use of a `workflow` input as described above.